Watch docs

eatkins opened the pull request:
I made these changes in response to writing documentation for the new watch apis.

eatkins committed:
revert doc

eatkins committed:
Remove leading semicolon from multi command help

eatkins committed:
Move watch default settings

I had previously set some of the watch settings at the global level and
some at the project level. While writing documentation for the new watch
subsystem, I realized that the defaults should be set globally so that
they can be overridden at the ThisBuild level.

I also moved watchTriggers to sbt.nio.Keys. It was an oversight that it
wasn't moved there in a5cefd45be77c4f7eb257a751d26c7ea66b9fdc9.

eatkins committed:
Use logger instead of directly printing to System.err

eatkins committed:
Fix parameters in watchTriggeredMessage

eatkins committed:
Simplify watch callbacks

While writing documentation for the watch subsystem, I realized that
it's awkward to configure watch to clear the screen before task
evaluation. To make this easier, I added a setting watchBeforeCommand
which is an arbitrary function that will run before the watch process
evaluates the command(s).

I also added helper functions for adding clear screen functionality.

I also realized that we weren't using the watchOnEnter or
watchOnExit callbacks anywhere. I had added these to support setting up
some state before watch starts and cleaning it up before it exits for
plugin authors. It makes sense to remove that functionality for 1.3.0
and only if a need presents itself re-add it in a later version of sbt.

I also made a few apis private[sbt] that I'm not sure about. Writing
documentation made me realize that some of these are redundant and/or
not ready for general consumption.

eatkins committed:
Add sbt.nio.Watch to default imports

I realized writing documentation that it is a pain to not have Watch
available in build.sbt with an import.

eatkins committed:
Switch private[sbt] status of Reload objects

The Reload exception that I added in the sbt package really wasn't
intended to be public. It's only meant to be used by
checkMetaBuildSources, which the users shouldn't override. I put it in
the top package though because I wanted it to be next to FullReload. I
also am not sure why the Reload object in Watch was private[sbt], but
while writing documentation, I realized that users couldn't access it.

eed3si9n commented on main/src/main/scala/sbt/nio/Keys.scala:
```diff
@@ -82,16 +82,13 @@ object Keys {
     "Callback to invoke if an event is triggered in a continuous build by one of the files matching an fileInput glob for the task and its transitive dependencies"
   ).withRank(DSetting)
-  val watchOnIteration = settingKey[Int => Watch.Action](
+  val watchOnIteration = settingKey[(Int, String, Seq[String]) => Watch.Action](
```
I didn't notice String in other keys before. Could this be typed to something like ProjectRef?

eatkins committed:
Change signature of pre watch methods

It makes the method parameters more clear if we pass in the ProjectRef
rather than the project name. We also don't lose information.
